src/Nethermind/Nethermind.JsonRpc.Connectors/Client/BasicJsonRpcClient.cs

```diff
@@ -70,6 +70,10 @@ public async Task<T> Post<T>(string method, params object[] parameters)
             {
                 throw;
             }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
```

Comment:
Don't do that, instead do `catch (Exception e) where (e is not TaskCanceledException)`

Reply:
done